src/core/comm/src/4C_comm_mpi_utils.hpp

```diff
@@ -28,113 +27,131 @@ FOUR_C_NAMESPACE_OPEN
 
 namespace Core::Communication
 {
+  /**
+   * Helper function during migration away from Epetra_Comm. Returns the MPI communicator from an
+   * Epetra_Comm object.
+   */
+  MPI_Comm unpack_epetra_comm(const Epetra_Comm& comm);
+
+  /**
+   * Helper function during migration away from Epetra_Comm. Returns the MPI_Comm @p comm wrapped in
+   * an Epetra_Comm object.
+   *
+   * @note Epetra_Comm is a virtual interface and it is not trivial to create an instance on the fly
+   * and return it here due to the lack of value semantics. Trilinos functions often expect a
+   * `const Epetra_Comm&`. Therefore, this function has to maintain a global cache of Epetra_Comm
+   * objects for the respective MPI communicator. They are kept alive for the duration of the
+   * program. Don't try to create this object yourself, as you will likely run into issues with
+   * dangling references.
+   */
+  const Epetra_Comm& as_epetra_comm(MPI_Comm comm);
+
```

Comment:
This is the only non-trivial change in this PR. I added these two helpers to convert between MPI_Comm and Epetra_Comm in a few places.